examples/shader/shader_material_wesl.rs

```diff
@@ -0,0 +1,126 @@
+//! A shader that uses the WESL shading language.
+
+use bevy::{
+    asset::{load_internal_asset, weak_handle},
+    pbr::{MaterialPipeline, MaterialPipelineKey},
+    prelude::*,
+    reflect::TypePath,
+    render::{
+        mesh::MeshVertexBufferLayoutRef,
+        render_resource::{
+            AsBindGroup, RenderPipelineDescriptor, ShaderDefVal, ShaderRef,
+            SpecializedMeshPipelineError,
+        },
+    },
+};
+
+/// This example uses shader source files from the assets subdirectory
+const FRAGMENT_SHADER_ASSET_PATH: &str = "shaders/custom_material.wesl";
+/// An example utility shader that is used by the custom material
+pub const UTIL_SHADER_HANDLE: Handle<Shader> = weak_handle!("748706a1-969e-43d4-be36-74559bd31d23");
+
+fn main() {
+    App::new()
+        .add_plugins((
+            DefaultPlugins,
+            MaterialPlugin::<CustomMaterial>::default(),
+            CustomMaterialPlugin,
+        ))
+        .add_systems(Startup, setup)
+        .add_systems(Update, update)
+        .run();
+}
+
+/// A plugin that loads the custom material shader
+pub struct CustomMaterialPlugin;
+
+impl Plugin for CustomMaterialPlugin {
+    fn build(&self, app: &mut App) {
+        load_internal_asset!(
```

Comment:
Is this necessary?

Reply:
Right now, yes, because this is using the asset system for all shader imports, but my next goal is to evaluate what it would look like to use crate/filesystem based lookups.